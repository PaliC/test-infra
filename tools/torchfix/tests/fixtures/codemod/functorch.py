import torch
import functorch

batch_size, feature_size = 3, 5
weights = torch.randn(feature_size, requires_grad=True)

def model(feature_vec):
    # Very simple linear model with activation
    return feature_vec.dot(weights).relu()

examples = torch.randn(batch_size, feature_size)
result = functorch.vmap(model)(examples)
print(result)

# Non-runnable, just to check name changes.
def f():
    functorch.vmap()
    functorch.grad()
    functorch.vjp()
    functorch.jvp()
    functorch.jacrev()
    functorch.jacfwd()
    functorch.hessian()
    functorch.functionalize()
