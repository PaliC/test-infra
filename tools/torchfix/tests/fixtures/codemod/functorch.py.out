import torch
import functorch

batch_size, feature_size = 3, 5
weights = torch.randn(feature_size, requires_grad=True)

def model(feature_vec):
    # Very simple linear model with activation
    return feature_vec.dot(weights).relu()

examples = torch.randn(batch_size, feature_size)
result = torch.func.vmap(examples)
print(result)

# Non-runnable, just to check name changes.
def f():
    torch.func.vmap()
    torch.func.grad()
    torch.func.vjp()
    torch.func.jvp()
    torch.func.jacrev()
    torch.func.jacfwd()
    torch.func.hessian()
    torch.func.functionalize()
